.github/workflows/on_build_do_test.yml

```diff
@@ -0,0 +1,210 @@
+
+jobs:
+  get-info:
+      name: Get Origin Workflow Info
+      runs-on: ubuntu-latest
+      outputs:
+         sourceHeadSha: ${{ github.event.workflow_run.head_sha }}
+         pullRequestNumber: ${{ steps.workflow-run-info.outputs.pullRequestNumber }}
+         sourceHeadBranch: ${{ steps.workflow-run-info.outputs.pullRequestLabels }}
+         sourceEvent: ${{ github.event.workflow_run.event }}
+      steps:
+         - id: workflow-run-info
+           uses: rigazilla/infinispan/.github/actions/get-origin-info@main
+           with:
+            head_sha: ${{ github.event.workflow_run.head_sha }}
+            head_branch: ${{ github.event.workflow_run.head_branch }}
+            event: ${{ github.event.workflow_run.event }}
+
+  ci-build-test-pr:
+    needs: get-info
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    name: Maven Test
+    runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: "-Xmx1500m -XX:+HeapDumpOnOutOfMemoryError"
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - run: echo !!!RUNNING PR!!!
+      - uses: actions/checkout@v4
+
+      - name: Setup Java
+        uses: ./.github/actions/setup-java
+
+      - uses: graalvm/setup-graalvm@v1
+        with:
+          java-version: 21
+          distribution: 'graalvm'
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract Maven Artifacts
+        uses: ./.github/actions/extract-maven-artifacts
+        with:
+         run_id: ${{ github.event.workflow_run.id }}
+
+      - name: Download Infinispan Server
+        uses: actions/download-artifact@v4
+        with:
+          name: infinispan-dist
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ github.token }}
+
+      - name: Download Server Version
+        uses: actions/download-artifact@v4
+        with:
+          name: server-version
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ github.token }}
+
+      - name: Get server version
+        id: server_version
+        uses: juliangruber/read-file-action@v1
+        with:
+           path: ./server-version.txt
+
+      - name: Extract Infinispan Source
+        run: |
+          mkdir test_dir && cd test_dir
+          unzip ${GITHUB_WORKSPACE}/infinispan-${{ steps.server_version.outputs.content }}-src.zip
+          mv infinispan-${{ steps.server_version.outputs.content }}-src infinispan
+
+      - name: Maven Test
+        shell: bash
+        run: >
+            cd test_dir/infinispan &&
+            ./mvnw verify -s maven-settings.xml -V -B -e -DrerunFailingTestsCount=2
+            -Dmaven.test.failure.ignore=true -Dansi.strip=true -Pnative -pl commons/all
+
+      - name: Archive commit sha PR
+        if: >
+          (success() || failure()) &&
+          needs.get-info.outputs.sourceEvent == 'pull_request'
+        run: |
+           echo -n ${{ needs.get-info.outputs.sourceHeadSha }} > github-sha.txt
+
+      - name: Archive commit sha push
```

Comment:
I don't think we need to create these sha files any more?

Reply:
is needed to attach the number-of-test-ran check to the correct commit:
https://github.com/rigazilla/infinispan/blob/4f736c33aee2da21356b3772a7ac2fb933533bdf/.github/workflows/on_test_publish_surefire_report.yml#L86